extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,152 @@
 import {Services} from '#service';
 
+import {user} from '#utils/log';
+
 import {getElementConfig} from 'extensions/amp-story/1.0/request-utils';
 
+import {assertHttpsUrl} from '../../../src/url';
 import {
   Action,
   ShoppingConfigDataDef,
 } from '../../amp-story/1.0/amp-story-store-service';
 
+/** @const {!Object<!Object<string, !Array<function>>>} */
+export const VALIDATION_OBJECTS = {
+  'aggregateRating': {
+    'ratingValue': [validateRequired, validateNumber],
+    'reviewCount': [validateRequired, validateNumber],
+    'reviewUrl': [validateRequired, validateURLs],
+  },
+  'productValidationConfig': {
+    /* Required Attrs */
+    'productUrl': [validateRequired, validateString],
+    'productId': [validateRequired, validateString],
+    'productTitle': [validateRequired, validateString],
+    'productBrand': [validateRequired, validateString],
+    'productPrice': [validateRequired, validateNumber],
+    'productImages': [validateRequired, validateURLs],
```

Comment:
This is now an object with `url` and `alt` entries. I think it might need to be changed to `validateObject`.

Reply:
That won't quite work. ProductImages is an Array of objects, not an object. I'll have to think of a workaround.

Reply:
Found an easy way to do it: keep using `validateURLs`, but only check for the alt attribute using `validateString` if the field is `productImages`.

```
 if (field === 'productImages') {
      validateString('productImages alt', url.alt);
 }
```

Reply:
+1 Lets try to avoid putting key specific logic in this function. If we correct the params type signature it would now be something like `Array<string>|string|Object{url, alt}` which seems like its doing too much. Agree that using `createValidateConfig` seems like the right approach for `productImages` obj and maybe we should introduce a `validateArray` as well

Reply:
As per Caleb's comment in https://github.com/ampproject/amphtml/pull/37474#discussion_r816292932 ,
I have included Array of Object validation for productImages in createValidateConfigArray